src/spawn_stubs.c

```diff
@@ -1,5 +1,25 @@
 #define _GNU_SOURCE
 
+/* Must come before any other caml/ headers are included */
+#define CAML_INTERNALS
+
+#ifdef _WIN32
```

Comment:
Instead of changing the `dune` file:
```suggestion
#ifdef _WIN32

#define UNICODE
#define _UNICODE

```
Based on a comment by @emillon in https://github.com/ocaml/dune/pull/10212#issuecomment-1980368796

Reply:
My only concern is that `UNICODE` and `_UNICODE` must either be defined everywhere or nowhere (OCaml headers can pull in `windows.h` or parts of it as well, ending up with a similar problem to that which we have already had with `CAML_INTERNALS`).

The other benefit of putting it in flags is that any additional C files correctly inherit it, but that's really never likely to apply here. Given that we already have the `_GNU_SOURCE` definition (which is where it should go - i.e. right at the top).

I don't mind, though - how strong opinions against doing it "philosophically" correctly, if more complicatedly, in the `dune` file?

Reply:
I agree it is better to do it in the `dune` file.